crates/common/mqtt_channel/src/config.rs

```diff
@@ -48,6 +51,34 @@ pub struct Config {
     ///
     /// Default: None
     pub last_will_message: Option<Message>,
+
+    /// With first message on connection
+    ///
+    /// Default: None
+    pub initial_message: Option<InitMessageFn>,
+}
+
+#[derive(Clone)]
+pub struct InitMessageFn {
+    initfn: Arc<Box<dyn Fn() -> Message + Send + Sync>>,
+}
+
+impl InitMessageFn {
+    pub fn new(call_back: impl Fn() -> Message + Sync + Send + 'static) -> InitMessageFn {
+        InitMessageFn {
+            initfn: Arc::new(Box::new(call_back) as Box<dyn Fn() -> Message + Sync + Send>),
+        }
+    }
+
+    pub fn call(&self) -> Message {
+        (*self.initfn)()
+    }
+}
+
+impl Debug for InitMessageFn {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Calling init_message function")
```

Comment:
This is not a call to the function, but the function itself.

```suggestion
        write!(f, "Init message creation function")
```